src/tracer/MOM_offline_control.F90

```diff
@@ -0,0 +1,619 @@
+    is  = G%isc ; ie  = G%iec ; js  = G%jsc ; je  = G%jec ; nz = GV%ke
+
+    ! Calculate top and bottom fluxes from ea and eb. Note the explicit negative signs
+    ! to enforce the positive out convention
+    k = 1
+    do j=js-1,je+1 ; do i=is-1,ie+1
+      top_flux(i,j,k) = -ea(i,j,k)
+      bottom_flux(i,j,k) = -(eb(i,j,k)-ea(i,j,k+1))
+    enddo ; enddo
+
+    do k=2, nz-1 ; do j=js-1,je+1 ; do i=is-1,ie+1
+      top_flux(i,j,k) = -(ea(i,j,k)-eb(i,j,k-1))
+      bottom_flux(i,j,k) = -(eb(i,j,k)-ea(i,j,k+1))
+    enddo ; enddo ; enddo
+
+    k=nz
+    do j=js-1,je+1 ; do i=is-1,ie+1
+      top_flux(i,j,k) = -(ea(i,j,k)-eb(i,j,k-1))
+      bottom_flux(i,j,k) = -eb(i,j,k)
+    enddo ; enddo
+
+
+    ! Calculate sum of positive fluxes (negatives applied to enforce convention)
+    ! in a given cell and scale it back if it would deplete a layer
+    do k = 1, nz ; do j=js-1,je+1 ; do i=is-1,ie+1
+
+      hvol = h_pre(i,j,k)*G%areaT(i,j)
+      pos_flux  = max(0.0,-uh(I-1,j,k)) + max(0.0, -vh(i,J-1,k)) + &
+        max(0.0, uh(I,j,k)) + max(0.0, vh(i,J,k)) + &
+        max(0.0, top_flux(i,j,k)*G%areaT(i,j)) + max(0.0, bottom_flux(i,j,k)*G%areaT(i,j))
+
+      if (pos_flux>hvol .and. pos_flux>0.0) then
+        scale_factor = ( hvol )/pos_flux*max_off_cfl
+      else ! Don't scale
+        scale_factor = 1.0
+      endif
+
+      ! Scale horizontal fluxes
+      if (-uh(I-1,j,k)>0) uh(I-1,j,k) = uh(I-1,j,k)*scale_factor
+      if (uh(I,j,k)>0)    uh(I,j,k)   = uh(I,j,k)*scale_factor
+      if (-vh(i,J-1,k)>0) vh(i,J-1,k) = vh(i,J-1,k)*scale_factor
+      if (vh(i,J,k)>0)    vh(i,J,k)   = vh(i,J,k)*scale_factor
+
+      if (k>1 .and. k<nz) then
+      ! Scale interior layers
+        if(top_flux(i,j,k)>0.0) then
+          ea(i,j,k) = ea(i,j,k)*scale_factor
+          eb(i,j,k-1) = eb(i,j,k-1)*scale_factor
+        endif
+        if(bottom_flux(i,j,k)>0.0) then
+          eb(i,j,k) = eb(i,j,k)*scale_factor
+          ea(i,j,k+1) = ea(i,j,k+1)*scale_factor
+        endif
+      ! Scale top layer
+      elseif (k==1) then
+        if(top_flux(i,j,k)>0.0)    ea(i,j,k) = ea(i,j,k)*scale_factor
+        if(bottom_flux(i,j,k)>0.0) then
+          eb(i,j,k)   = eb(i,j,k)*scale_factor
+          ea(i,j,k+1) = ea(i,j,k+1)*scale_factor
+        endif
+      ! Scale bottom layer
+      elseif (k==nz) then
+        if(top_flux(i,j,k)>0.0) then
+          ea(i,j,k)   = ea(i,j,k)*scale_factor
+          eb(i,j,k-1) = eb(i,j,k-1)*scale_factor
+        endif
+        if (bottom_flux(i,j,k)>0.0) eb(i,j,k)=eb(i,j,k)*scale_factor
+      endif
+    enddo ; enddo ; enddo
+
+  end subroutine limit_mass_flux_3d
+
+end module MOM_offline_transport
+
+!  Instructions for running passive tracers offline in MOM6
+!  Contact: Andrew Shao (andrew.shao@noaa.gov)
+!  Last modified: 7 October 2016
+!
+!  ----
+!  QUICK-START
+!  ----
+!  1) Follow instructions to compile MOM6 executables using ice_ocean_SIS2 and ocean_only drivers
```

Comment:
I know that there are some references to directory structure in the code elsewhere (I am removing them) which have set a bad precedent but the directory structure of MOM6-examples should not be referenced in the source code since these repositories are independent. This quick-start section is also in the README.offline in the associated PR NOAA-GFDL/MOM6-examples#110 which is where it belongs. Keep the rest and doxygenize (`!!`).